fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -91,18 +99,73 @@ public void analyzeIgnoreSlot(Expr expression, AnalyzeState analyzeState, Scope
         bottomUpAnalyze(visitor, expression, scope);
     }
 
-    private void bottomUpAnalyze(Visitor visitor, Expr expression, Scope scope) {
-        for (Expr expr : expression.getChildren()) {
+    private boolean isHighOrderFunction(Expr expr) {
+        if (expr instanceof FunctionCallExpr) {
+            // expand this in the future.
+            if (((FunctionCallExpr) expr).getFnName().getFunction().equalsIgnoreCase(FunctionSet.ARRAY_MAP)) {
+                return true;
+            } else if (((FunctionCallExpr) expr).getFnName().getFunction().equalsIgnoreCase(FunctionSet.TRANSFORM)) {
+                // transform just a alias of array_map
+                ((FunctionCallExpr) expr).resetFnName("", FunctionSet.ARRAY_MAP);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // only high-order functions can use lambda functions.
+    void analyzeHighOrderFunction(Visitor visitor, Expr expression, Scope scope) {
+        if (!isHighOrderFunction(expression)) {
+            throw new SemanticException("Lambda Functions can only be used in supported high-order functions.");
+        }
+        int childSize = expression.getChildren().size();
+        // move the lambda function to the first if it is at the last.
```

Comment:
What if  lambda function is in the middle, why need to specially handle last position?

Reply:
lambda functions can only lay at the first (ck) or last (presto) position.